crates/core_simd/src/vector.rs

```diff
@@ -176,34 +176,62 @@ where
         unsafe { &mut *(self as *mut Self as *mut [T; N]) }
     }
 
+    /// Load a vector from an array of `T`.
+    ///
+    /// This function is necessary since `repr(simd)` has padding for non-power-of-2 vectors (at the time of writing).
+    /// With padding, `read_unaligned` will read past the end of an array of N elements.
+    ///
+    /// # Safety
+    /// Reading `ptr` must be safe, as if by `<*const [T; N]>::read_unaligned`.
+    const unsafe fn load(ptr: *const [T; N]) -> Self {
+        let mut tmp = core::mem::MaybeUninit::uninit();
+        // SAFETY: `Simd<T, N>` always contains `N` elements of type `T`.  It may have padding
+        // which does not need to be initialized.  The safety of reading `ptr` is ensured by the
+        // caller.
+        unsafe {
+            core::ptr::copy_nonoverlapping(ptr, tmp.as_mut_ptr() as *mut _, 1);
```

Comment:
ymmv nit: If you're going to `as` with an `_`, might as well use `cast`:
```suggestion
            core::ptr::copy_nonoverlapping(ptr, tmp.as_mut_ptr().cast(), 1);
```

Reply:
This needs `::<Self>` on `MaybeUninit` to avoid a type inference issue:
```
error[E0699]: the type of this value must be known to call a method on a raw pointer on it
   --> crates/core_simd/src/vector.rs:192:66
    |
192 |             core::ptr::copy_nonoverlapping(ptr, tmp.as_mut_ptr().cast(), 1);
    |
    ```

Reply:
Ah, I see.

(TBH, I like the `::<Self>` since in unsafe code my usual "it's fine, if it infers weird rust will tell me" is less reliable.)

Reply:
Agreed, I thought it was more clear in the end. I always forget about `cast` too...